File: 2387/Dentista.py
n = int(input())

horarios = []
for i in range(n):

    x, y = [int(x) for x in input().split()]
    horarios.append((x,y))


horarios.sort(key = lambda x:x[1])

max = 1
j = 0

for i in range(0, len(horarios)):
    
    if horarios[i][0] >= horarios[j][1]:
        max+=1
        j = i

print(max)

File: 2387/Dentista.cpp
#include <bits/stdc++.h>
using namespace std;

bool sortbysec(const pair<int,int> &a,
              const pair<int,int> &b)
{
    return (a.second < b.second);
}

int main(){

    int n; cin >> n;
    vector<pair<int,int>> horarios;
    int x, y;

    while (n--){

        cin >> x >> y;

        horarios.push_back(make_pair(x,y));
    }

    sort(horarios.begin(), horarios.end(), sortbysec);
    int max = 1;
    int j = 0;
    for (int i = 1; i < horarios.size(); i++){
        if (horarios[i].first >= horarios[j].second){
            max++;
            j = i;
        }
    }

    cout << max << endl;
}
